libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -333,9 +338,71 @@ - (SFRestRequest *)requestForSearchResultLayout:(NSString*)objectList {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (SFRestRequest *request in requests) {
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.path, [self toQueryString:request.queryParams]];
+        }
+        // queryParams belongs in body
+        else {
+            requestJson[@"url"] = request.path;
+            requestJson[@"richInput"] = request.queryParams;
+        }
+        [requestsArrayJson addObject:requestJson];
+    }
+    NSMutableDictionary<NSString *, id> *batchRequestJson = [NSMutableDictionary new];
+    batchRequestJson[@"batchRequests"] = requestsArrayJson;
+    batchRequestJson[@"haltOnError"] = [NSNumber numberWithBool:haltOnError];
+
+    NSString *path = [NSString stringWithFormat:@"/%@/composite/batch", self.apiVersion];
+
+    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:batchRequestJson];
+}
+
+- (SFRestRequest *) compositeRequest:(NSArray<SFRestRequest*>*) requests refIds:(NSArray<NSString*>*)refIds allOrNone:(BOOL) allOrNone {
+    NSMutableArray *requestsArrayJson = [NSMutableArray new];
+    for (int i=0; i<requests.count; i++) {
+        SFRestRequest *request = requests[i];
+        NSString *refId = refIds[i];
+        NSMutableDictionary<NSString *, id> *requestJson = [NSMutableDictionary new];
+        requestJson[@"referenceId"] = refId;
+        requestJson[@"method"] = [SFRestRequest httpMethodFromSFRestMethod:request.method];
+        // queryParams belong in url
+        if (request.method == SFRestMethodGET || request.method == SFRestMethodDELETE) {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@%@", request.endpoint, request.path, [self toQueryString:request.queryParams]];
+        }
+        // queryParams belongs in body
+        else {
+            requestJson[@"url"] = [NSString stringWithFormat:@"%@%@", request.endpoint, request.path];
+            requestJson[@"body"] = request.queryParams;
+        }
+        [requestsArrayJson addObject:requestJson];
+    }
+    NSMutableDictionary<NSString *, id> *compositeRequestJson = [NSMutableDictionary new];
+    compositeRequestJson[@"compositeRequest"] = requestsArrayJson;
+    compositeRequestJson[@"allOrNone"] = [NSNumber numberWithBool:allOrNone];
 
-    return [httpDateFormatter stringFromDate:date];
+    NSString *path = [NSString stringWithFormat:@"/%@/composite", self.apiVersion];
+
+    return [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:compositeRequestJson];
 }
+
+- (NSString *)toQueryString:(NSDictionary *)components {
```

Comment:
Could we switch this util method to something like this instead:
```
NSURLComponents *components = [NSURLComponents componentsWithString:fullUrl];

        // Adds query parameters to the request if any are set.
        if (self.queryParams) {
            NSMutableArray<NSURLQueryItem *> *queryItems = [[NSMutableArray alloc] init];
            for (NSString *key in self.queryParams.allKeys) {
                if (key != nil) {
                    NSURLQueryItem *query = [NSURLQueryItem queryItemWithName:key value:self.queryParams[key]];
                    [queryItems addObject:query];
                }
            }
            components.queryItems = queryItems;
        }
        self.request = [[NSMutableURLRequest alloc] initWithURL:components.URL];
```

Using `NSMutableString` doesn't URL encode the query params.